mapeditor/playersettings.cpp

```diff
@@ -0,0 +1,77 @@
+#include "StdInc.h"
+#include "playersettings.h"
+#include "ui_playersettings.h"
+#include "playerparams.h"
+#include "mainwindow.h"
+
+PlayerSettings::PlayerSettings(MapController & ctrl, QWidget *parent) :
+	QDialog(parent),
+	ui(new Ui::PlayerSettings),
+	controller(ctrl)
+{
+	ui->setupUi(this);
+	show();
+
+	int players = 0;
+	for(auto & p : controller.map()->players)
+	{
+		if(p.canAnyonePlay())
+		{
+			paramWidgets.push_back(new PlayerParams(controller, players));
+			ui->playersLayout->addWidget(paramWidgets.back());
+			++players;
+		}
+	}
+
+	if(players < 2)
```

Comment:
Could you please replace `2` with a named constant?

Reply:
could you please suggest the name?

Reply:
I'm not really sure. Is it 2 because there must be 2 minimum players? In that case, maybe `NUM_MIN_PLAYERS`? I can't fully tell though why 2 is used a lot in this file.

Reply:
I don't understand the meaning of 2 either

Reply:
single player? There is another approach. auto singlePlayer = players < 2;

Reply:
So there is a lot of this 2 everywhere so need a constant)

Reply:
Btw MIN_NUM_PLAYERS = 1 might be possible and use players <= MIN_NUM_PLAYERS